
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
        <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>itcrispr — An interface to assemble and track progress of CRISPR sequence experiments in BioNumerics &#8212; BioNumerics Scripts 1.0 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="misc — Miscellaneous scripts" href="misc.html" />
    <link rel="prev" title="itquery — An interface to query ItemTracker from BioNumerics" href="itquery.html" />

   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

<link href="_static/buttons-min.css" rel="stylesheet" type="text/css">
<link href="_static/custom.css" rel="stylesheet" type="text/css">

  </head><body>

  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="itcrispr-an-interface-to-assemble-and-track-progress-of-crispr-sequence-experiments-in-bionumerics">
<h1>itcrispr — An interface to assemble and track progress of CRISPR sequence experiments in BioNumerics<a class="headerlink" href="#itcrispr-an-interface-to-assemble-and-track-progress-of-crispr-sequence-experiments-in-bionumerics" title="Permalink to this headline">¶</a></h1>
<p>Version: <code class="docutils literal notranslate"><span class="pre">0.1.4</span></code></p>
<div class="section" id="about">
<h2>About<a class="headerlink" href="#about" title="Permalink to this headline">¶</a></h2>
<p>A program to assist in the analysis of CRISPR sequence experiments in BioNumerics. Assembly is done using predefined primers. The finished status of each primer
is saved within the program.</p>
</div>
<div class="section" id="setup-and-configuration">
<h2>Setup and configuration<a class="headerlink" href="#setup-and-configuration" title="Permalink to this headline">¶</a></h2>
<div class="section" id="install-python-2-6">
<h3>Install Python 2.6<a class="headerlink" href="#install-python-2-6" title="Permalink to this headline">¶</a></h3>
<p>Download (x86 version): <a class="reference external" href="http://www.python.org/download/releases/2.6.6/">http://www.python.org/download/releases/2.6.6/</a></p>
</div>
<div class="section" id="install-pyqt">
<h3>Install PyQt<a class="headerlink" href="#install-pyqt" title="Permalink to this headline">¶</a></h3>
<p>Download (x86 version for Python 2.6): <a class="reference external" href="http://www.riverbankcomputing.co.uk/software/pyqt/download">http://www.riverbankcomputing.co.uk/software/pyqt/download</a></p>
</div>
<div class="section" id="install-itcrispr">
<h3>Install itcrispr<a class="headerlink" href="#install-itcrispr" title="Permalink to this headline">¶</a></h3>
<p>Clone repository</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">git</span><span class="nd">@github</span><span class="o">.</span><span class="n">com</span><span class="p">:</span><span class="n">vkvn</span><span class="o">/</span><span class="n">bionumerics_scripts</span><span class="o">.</span><span class="n">git</span>
</pre></div>
</div>
<p>Build and install</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="n">bionumerics_scripts</span><span class="o">/</span><span class="n">itcrispr</span>
<span class="n">python</span> <span class="n">setup</span><span class="o">.</span><span class="n">py</span> <span class="n">build</span>
<span class="n">python</span> <span class="n">setup</span><span class="o">.</span><span class="n">py</span> <span class="n">install</span>
</pre></div>
</div>
</div>
<div class="section" id="configure-script">
<h3>Configure script<a class="headerlink" href="#configure-script" title="Permalink to this headline">¶</a></h3>
<ol class="arabic">
<li><p class="first">Change configuration in</p>
<p><code class="docutils literal notranslate"><span class="pre">C:\Python26\Lib\site-packages\itcrispr\itcrispr.cfg</span></code></p>
<p><code class="docutils literal notranslate"><span class="pre">DATABASES</span></code>, <code class="docutils literal notranslate"><span class="pre">EXPERIMENTS</span></code> and <code class="docutils literal notranslate"><span class="pre">PRIMER_PATH</span></code> are all <strong>required</strong> for
the program to work.</p>
<p><code class="docutils literal notranslate"><span class="pre">DATABASES</span></code></p>
<blockquote>
<div><p>The database name as in BioNumerics. Multiple databases can be specified
as comma-separated values. This option is to prevent the script from being
executed accidentally in other databases.</p>
</div></blockquote>
<p><code class="docutils literal notranslate"><span class="pre">EXPERIMENTS</span></code></p>
<blockquote>
<div><p>List of CRISPR sequence experiments - comma-separated.</p>
</div></blockquote>
<p><code class="docutils literal notranslate"><span class="pre">PRIMER_PATH</span></code></p>
<blockquote>
<div><p>PATH to the directory containing primer files in text format. File
extension should be <code class="docutils literal notranslate"><span class="pre">.txt</span></code> and file should contain only the
sequence (FASTA and other formats not supported).</p>
</div></blockquote>
</li>
</ol>
</div>
</div>
<div class="section" id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h2>
<ol class="arabic">
<li><p class="first">Select entries in BioNumerics. If no selection is made, program runs through
all entries in the database.</p>
</li>
<li><p class="first">Run script using the <code class="docutils literal notranslate"><span class="pre">Script</span> <span class="pre">--&gt;</span> <span class="pre">Run</span> <span class="pre">script</span> <span class="pre">from</span> <span class="pre">file</span></code>
option in BioNumerics. Navigate to the <code class="docutils literal notranslate"><span class="pre">C:\Python26\Scripts</span></code> folder and
select <code class="docutils literal notranslate"><span class="pre">run_itcrispr.py</span></code></p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">As easier method to run the script would be to add a button to the toolbar.
Steps for doing this is described in <a class="reference internal" href="#add-a-button-to-the-toolbar">Add a button to the toolbar</a>.</p>
</div>
</li>
<li><p class="first">An assembly will be done using the CRISPR sequences and primers. A dialog
window will then be displayed with the list of primers that aligned with the
sequences. Double-click on a primer to navigate to that region in the assembler.</p>
<img alt="_images/itcrispr_mainwindow.jpg" src="_images/itcrispr_mainwindow.jpg" />
</li>
<li><p class="first">If you consider an experiment for an entry as finished, click the
<strong>Save as Finished</strong> button. No primer should be checked. Use the
<strong>Select None</strong> button to deselect all primers.</p>
</li>
<li><p class="first">To save an experiment as unfinished, check the primers for regions that
need to be repeated and click the <strong>Save as Unfinished</strong> button.</p>
</li>
<li><p class="first">If you made some changes in the assembler window, use the <strong>Recalculate</strong>
button to redo the assembly.</p>
</li>
<li><p class="first">Use the <strong>Skip</strong> button to redo this entry at a later time.</p>
</li>
<li><p class="first">Use the <strong>Export Database</strong> button to export the entire database to a <code class="docutils literal notranslate"><span class="pre">CSV</span></code>
file.</p>
</li>
<li><p class="first"><strong>Stop</strong> button will terminate the program.</p>
</li>
</ol>
</div>
<div class="section" id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>If the experiment for an entry is saved as finished, it will not be displayed
in subsequent runs.</li>
<li>Primer sequences are read, an assembly is done after changing the minimum
score to the smallest length of primer.</li>
<li>Primers selected previously appear checked when the window is shown.</li>
<li>Closing the window skips to the next experiment. Use the <strong>Stop</strong> button to
exit the program.</li>
<li>Sequence approval status in BioNumerics is not taken into consideration for
deciding which entries to check. Sequences are made approved if they are
saved as finished using the program.</li>
</ul>
<div class="section" id="add-a-button-to-the-toolbar">
<span id="add-button"></span><h3>Add a button to the toolbar<a class="headerlink" href="#add-a-button-to-the-toolbar" title="Permalink to this headline">¶</a></h3>
<p><strong>This needs to be done whenever the program is updated!</strong></p>
<ol class="arabic">
<li><p class="first">Copy</p>
<p><code class="docutils literal notranslate"><span class="pre">C:\Python26\Lib\site-packages\itcrispr\itcrispr-menu.bns</span></code> to</p>
<p>the <cite>scripts home</cite> directory under your BioNumerics Home directory. For example,</p>
<p><code class="docutils literal notranslate"><span class="pre">C:\Users\vimal\BioNumerics\Data\Scripts</span> <span class="pre">Home\</span></code></p>
</li>
<li><p class="first">Load this file in the <strong>BNS</strong> Script editor</p>
<p><code class="docutils literal notranslate"><span class="pre">Script</span> <span class="pre">--&gt;</span> <span class="pre">BNS</span> <span class="pre">Script</span> <span class="pre">editor</span></code></p>
<p>followed by</p>
<p><code class="docutils literal notranslate"><span class="pre">File</span> <span class="pre">--&gt;</span> <span class="pre">Load</span> <span class="pre">Script</span></code></p>
</li>
<li><p class="first">Open <code class="docutils literal notranslate"><span class="pre">File</span> <span class="pre">--&gt;</span> <span class="pre">Attachments</span></code> and click <strong>Add new</strong></p>
</li>
<li><p class="first">Select <code class="docutils literal notranslate"><span class="pre">C:\Python26\Scripts\run_itcrispr.py</span></code> and click <strong>Open</strong>.
Enter <code class="docutils literal notranslate"><span class="pre">run_itcrispr</span></code> in the dialog that appears and click <strong>OK</strong>.</p>
</li>
<li><p class="first">Save the script.</p>
</li>
<li><p class="first">Restart BioNumerics. An icon <img alt="icon" src="_images/itcrispr_logo.png" />  should now be available in the toolbar. Click
this icon to launch the program.</p>
</li>
</ol>
</div>
<div class="section" id="add-or-remove-primer-sequences">
<h3>Add or remove primer sequences<a class="headerlink" href="#add-or-remove-primer-sequences" title="Permalink to this headline">¶</a></h3>
<p>Primer files are plain text files with just the sequence of the primer.</p>
<p>Add or remove primers to/from the location specified in the configuration
file - the <code class="docutils literal notranslate"><span class="pre">PRIMER_PATH</span></code> variable.</p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">BioNumerics Scripts</a></h1>






<p>
<iframe src="https://ghbtns.com/github-btn.html?user=vimalkvn&repo=bionumerics_scripts&type=watch&count=true&size=large&v=2"
  allowtransparency="true" frameborder="0" scrolling="0" width="200px" height="35px"></iframe>
</p>





<h3>Navigation</h3>


<ul>
<li><a class="reference internal" href="#">itcrispr — An interface to assemble and track progress of CRISPR sequence experiments in BioNumerics</a><ul>
<li><a class="reference internal" href="#about">About</a></li>
<li><a class="reference internal" href="#setup-and-configuration">Setup and configuration</a><ul>
<li><a class="reference internal" href="#install-python-2-6">Install Python 2.6</a></li>
<li><a class="reference internal" href="#install-pyqt">Install PyQt</a></li>
<li><a class="reference internal" href="#install-itcrispr">Install itcrispr</a></li>
<li><a class="reference internal" href="#configure-script">Configure script</a></li>
</ul>
</li>
<li><a class="reference internal" href="#usage">Usage</a></li>
<li><a class="reference internal" href="#notes">Notes</a><ul>
<li><a class="reference internal" href="#add-a-button-to-the-toolbar">Add a button to the toolbar</a></li>
<li><a class="reference internal" href="#add-or-remove-primer-sequences">Add or remove primer sequences</a></li>
</ul>
</li>
</ul>
</li>
</ul>



        </div>
      </div>
      <div class="clearer"></div>
    </div>

    <div class="footer">
      &copy;2013, Vimalkumar Velayudhan.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.7.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/itcrispr.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>